### PR TITLE
OPSIM-1176 : Add additional tools for comcam sv surveys

### PR DESCRIPTION
**Pull request by rhiannonlynne**
Primarily add more convenience functions to the 'example' directory.

### CODE REVIEW

**Review by yoachim — APPROVED** [review]

File: rubin_scheduler/scheduler/example/comcam_sv_scheduler_example.py [hunk 1]
```diff
@@ -0,0 +1,935 @@
...
+        ("DEEP_A0", 216, -12.5),  # DEEP Solar Systen
+        ("Rubin_SV_250_2", 250.0, 2.0),  # 250 High stellar densty, low extinction
+        ("Rubin_SV_300_-41", 300.0, -41.0),  # High stellar densty, low extinction
+        ("Rubin_SV_280_-48", 280.0, -48.0),  # High stellar densty, low extinction
+        ("DEEP_B0", 310, -19),  # DEEP Solar System
+        ("ELAIS_S1", 9.45, -44.0),  # ELAIS-S1 LSST DDF
+        ("XMM_LSS", 35.708333, -4.75),  # LSST DDF
+        ("ECDFS", 53.125, -28.1),  # ECDFS
+        ("COSMOS", 150.1, 2.1819444444444445),  # COSMOS
+        ("EDFS_A", 58.9, -49.315),  # EDFS_a
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/example/comcam_sv_scheduler_example.py:825]: I was using what Keith had, but this is a good point and I'll check in with him.

File: rubin_scheduler/scheduler/features/features.py [hunk 2]
```diff
@@ -319,10 +319,18 @@ def add_observation(self, observation, indx=None):
...
+
     Parameters
     ----------
     note : `str` (None)
         Only count observations that contain str in their note field
+
+
+    Notes
+    -----
+    add_observations_array may not work in the same manner as add_observation.
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/features/features.py:333]: Yeah .. it was just harder to test. I know better how to generate observations now, so I can probably write a test to check in on this now.  (I suspect there are other basis functions and features which have the same problem...)

File: rubin_scheduler/scheduler/sim_runner.py [hunk 3]
```diff
@@ -141,11 +140,14 @@ def sim_runner(
...
+                raise RuntimeError(
+                    "Scheduler has failed to provide a valid observation multiple times "
+                    f" at time ({observatory.mjd} from survey {scheduler.survey_index}."
+                )
             # if this is a first offence, might just be that targets set.
-            # Flush queue and get some new targets.
+            # Flush queue and try to get some new targets.
             scheduler.flush_queue()
-            mjd_last_flush = observatory.mjd + 0
+            mjd_last_flush = observatory.mjd
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/sim_runner.py:150]: I was assuming this was a float, but it's true sometimes things I expect to be floats are showing up as 1-element numpy arrays.

File: rubin_scheduler/scheduler/detailers/dither_detailer.py [hunk 4]
```diff
@@ -290,3 +295,91 @@ def __call__(self, observation_list, conditions):
...
+    def _generate_offsets_filter_change(self, filter_list, mjd, initial_offset):
+        """Generate a random camera rotation for each filter change
+        or add a small offset for each sequential observation.
+        """
+        mjd_hash = round(100 * (np.asarray(mjd).item() % 100))
+        rng = np.random.default_rng(mjd_hash * self.seed)
+
+        offsets = np.zeros(len(filter_list))
+        # offset = np.asarray(initial_offset).item()
+        # offsets[0] = offset
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/detailers/dither_detailer.py:345]: Removed.

File: rubin_scheduler/scheduler/schedulers/filter_scheduler.py [hunk 5]
```diff
@@ -33,6 +35,34 @@ def __call__(self, conditions):
...
+    """
+
+    def __init__(
+        self,
+        illum_bins=np.arange(0, 100 + 1, 25),
+        filter_sets=(("u", "g", "r"), ("g", "r", "i"), ("r", "i", "z"), ("i", "z", "y")),
+    ):
+        self.illum_bins = illum_bins
+        self.filter_sets = filter_sets
+
```
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/schedulers/filter_scheduler.py:52]: I'll update. This also is completely made up, as just a first thing to let Keith test a variety of filter use-cases, but it's good to make sure it won't fail obviously.

File: rubin_scheduler/scheduler/example/comcam_sv_scheduler_example.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/example/comcam_sv_scheduler_example.py:825]: I've made it update, for now, and in the future we could turn it off if needed instead.

File: rubin_scheduler/scheduler/features/features.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rhiannonlynne) [reply rubin_scheduler/scheduler/features/features.py]: Tested and consistent, now.

**Review by yoachim — APPROVED** [review]
All looks fine

**Pull request merged**